hazelcast/src/main/java/com/hazelcast/internal/ascii/TextCommandServiceImpl.java

```diff
@@ -372,6 +366,7 @@ public int size(String queueName) {
 
     @Override
     public void sendResponse(TextCommand textCommand) {
+        ((AbstractTextCommand) textCommand).beforeSendResponse(this);
```

Comment:
Can we declare this method in `TextCommand` so that this casting becomes unnecessary?

Reply:
Sure why not. Would you like to do it yourself after merging ? I can change my pr if you want also.

Reply:
Sure, I will do that in my branch.